### PR TITLE
Service: add additional options

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Adds some of the missing launchctl settings.

### CODE REVIEW

**Comment by carlocab** [comment]
Can we also have an option to override the manual command, just like with `plist_options manual:`? Sometimes the command you want to run manually is different. See, for example, [`emacs`](https://github.com/Homebrew/homebrew-core/blob/e9627d39cd78b2fb5ff2d3ee8e11e1de70e7c1a1/Formula/emacs.rb#L79-L101).

**Comment by SMillerDev** [comment]
Not as part of this PR iyam, but I think that should be possible.

File: Library/Homebrew/service.rb [hunk 1]
```diff
@@ -47,6 +47,30 @@ def working_dir(path = nil)
...
 
+    sig { params(path: T.nilable(T.any(String, Pathname))).returns(T.nilable(String)) }
+    def root_dir(path = nil)
+      case T.unsafe(path)
+      when nil
+        @root_dir
+      when String, Pathname
+        @root_dir = path.to_s
+      else
+        raise TypeError, "Service#root_dir expects a String"
```
Comment (Rylan12) [inline Library/Homebrew/service.rb:58]:
```suggestion
        raise TypeError, "Service#root_dir expects a String or Pathname"
```

Not necessary I guess but more accurate

File: Library/Homebrew/service.rb [hunk 2]
```diff
@@ -47,6 +47,30 @@ def working_dir(path = nil)
...
+
+    sig { params(path: T.nilable(T.any(String, Pathname))).returns(T.nilable(String)) }
+    def input_path(path = nil)
+      case T.unsafe(path)
+      when nil
+        @input_path
+      when String, Pathname
+        @input_path = path.to_s
+      else
+        raise TypeError, "Service#input_path expects a String"
```
Comment (Rylan12) [inline Library/Homebrew/service.rb:70]: Same here (see above comment)

```suggestion
        raise TypeError, "Service#input_path expects a String or Pathname"
```

**Review by Rylan12 — DISMISSED** [review]

**Review by Bo98 — APPROVED** [review]

**Review by Rylan12 — APPROVED** [review]

**Pull request merged**